modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -0,0 +1,480 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "PinMeshGenerator.h"
+
+#include <cmath>
+#include "MooseApp.h"
+#include "Factory.h"
+#include "libmesh/elem.h"
+
+registerMooseObject("ReactorApp", PinMeshGenerator);
+
+InputParameters
+PinMeshGenerator::validParams()
+{
+  auto params = MeshGenerator::validParams();
+
+  params.addRequiredParam<MeshGeneratorName>(
+      "reactor_params",
+      "The ReactorMeshParams MeshGenerator that is the basis for this component conformal mesh.");
+
+  params.addRequiredParam<subdomain_id_type>("pin_type",
+                                             "The integer ID for this pin type definition");
+
+  params.addRequiredRangeCheckedParam<Real>(
+      "pitch", "pitch>0.0", "The pitch for the outermost boundary polygon");
+
+  params.addRequiredRangeCheckedParam<unsigned int>(
+      "num_sectors", "num_sectors>0", "Number of azimuthal sectors in each quadrant");
+
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "ring_radii", "ring_radii>0.0", "Radii of major concentric circles");
+
+  params.addRangeCheckedParam<std::vector<Real>>(
+      "duct_halfpitch", "duct_halfpitch>0.0", "Apothem of the ducts");
+
+  params.addRangeCheckedParam<std::vector<unsigned int>>(
+      "mesh_intervals",
+      std::vector<unsigned int>{1},
+      "mesh_intervals>0",
+      "The number of meshing intervals for each region starting at the center (length(ring_radii) "
+      "+ "
+      "length(duct_halfpitch )+ 1)");
+
+  params.addParam<std::vector<std::vector<subdomain_id_type>>>(
+      "region_ids",
+      "IDs for each radial and axial zone for assignment of unique physical characteristics");
+
+  params.addParam<bool>("extrude",
+                        false,
+                        "Determines if this is the final step in the geometry construction"
+                        " and extrudes the 2D geometry to 3D. If this is true then this mesh "
+                        "cannot be used in further mesh building");
```

Comment:
```suggestion
                        "cannot be used in further mesh building in the Reactor workflow");
```
an alternative wording